deployment/ccip/view/v1_5/tokenadminregistry.go

```diff
@@ -39,19 +42,23 @@ func GenerateTokenAdminRegistryView(taContract *token_admin_registry.TokenAdminR
 
 // getAllConfiguredTokensPaginated fetches all configured tokens from the TokenAdminRegistry contract in paginated
 // manner to avoid RPC timeouts since the list of configured tokens can grow to be very large over time.
-func getAllConfiguredTokensPaginated(taContract *token_admin_registry.TokenAdminRegistry) ([]common.Address, error) {
-	startIndex := uint64(0)
-	allTokens := make([]common.Address, 0)
-	for {
-		fetchedTokens, err := taContract.GetAllConfiguredTokens(nil, startIndex, GetTokensPaginationSize)
+func getAllConfiguredTokensPaginated(taContract *token_admin_registry.TokenAdminRegistry) (map[common.Address]TokenDetails, error) {
+	tokenDetails := make(map[common.Address]TokenDetails)
+	allTokens, err := shared.GetSupportedTokens(taContract)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get supported tokens for tokenAdminRegistry %s: %w", taContract.Address().String(), err)
+	}
+	for _, token := range allTokens {
```

Comment:
nit: should we perform these concurrently? I have gotten feedback that changesets run slowly, especially in the context of test setup. Though these are just views, so not as critical.